libloki/api.js

```diff
@@ -63,36 +63,77 @@
     await outgoingMessage.sendToNumber(pubKey);
   }
 
-  async function sendPairingAuthorisation(secondaryDevicePubKey, signature) {
-    const pairingAuthorisation = new textsecure.protobuf.PairingAuthorisationMessage(
-      {
-        signature,
-        primaryDevicePubKey: textsecure.storage.user.getNumber(),
-        secondaryDevicePubKey,
-        type:
-          textsecure.protobuf.PairingAuthorisationMessage.Type.PAIRING_REQUEST,
-      }
+  function createPairingAuthorisationProtoMessage({
+    primaryDevicePubKey,
+    secondaryDevicePubKey,
+    requestSignature,
+    grantSignature,
+    type,
+  }) {
+    if (
+      !primaryDevicePubKey ||
+      !secondaryDevicePubKey ||
+      type === undefined ||
+      type === null
+    ) {
+      throw new Error(
+        'createPairingAuthorisationProtoMessage: pubkeys or type is not set'
+      );
+    }
+    if (requestSignature.constructor !== ArrayBuffer) {
```

Comment:
Why do you check for existence of grantSignature below but not requestSignature?

Reply:
Good point!
The idea is that the requestSignature should always be set, while the grantSignature is optional.
I better be checking if requestSignature is falsey in the condition above :) Will fix asap!